### PR TITLE
fix: skip already converted credentials

### PR DESCRIPTION
**Pull request by bzwei**
Any converted credentials are deleted. But if they are recreated by any customized script, duplicated credentials should be skipped.

AAP-22365: EDA create_initial_data command is no longer idempotent

### CODE REVIEW

File: src/aap_eda/core/management/commands/create_initial_data.py [hunk 1]
```diff
@@ -446,17 +450,26 @@ def _copy_registry_credentials(self):
...
                 "password": cred.secret.get_secret_value(),
             }
-            eda_cred = models.EdaCredential.objects.create(
+            eda_cred, created = models.EdaCredential.objects.get_or_create(
                 name=cred.name,
-                description=cred.description,
-                managed=False,
-                credential_type=cred_type,
-                inputs=inputs_to_store(inputs),
+                defaults={
```
Comment (mkanoor) [inline src/aap_eda/core/management/commands/create_initial_data.py:455]: @bzwei Will this defaults help if the inputs is changed between different releases of the Credential Type?

Comment (bzwei) [reply src/aap_eda/core/management/commands/create_initial_data.py:455]: defaults is used to create new credentials. The already created EdaCredential does not change.

File: src/aap_eda/core/management/commands/create_initial_data.py [hunk 2]
```diff
@@ -479,17 +492,26 @@ def _copy_scm_credentials(self):
...
+                models.Project.objects.filter(credential=cred).update(
+                    eda_credential=eda_cred, credential=None
+                )
+                cred.delete()
+            else:
+                warn_msg = (
+                    f"Git Credential {cred.name} already converted to "
+                    "EdaCredential. Skip the duplicated one."
+                )
+                LOGGER.warning(warn_msg)
```
[the remark this answers is not in the document]
Comment (bzwei) [reply src/aap_eda/core/management/commands/create_initial_data.py:514]: an info level?

[the remark this answers is not in the document]
Comment (Alex-Izquierdo) [reply src/aap_eda/core/management/commands/create_initial_data.py:514]: Yeah, more appropriate IMO

File: src/aap_eda/core/management/commands/create_initial_data.py [hunk 3]
```diff
@@ -446,17 +450,26 @@ def _copy_registry_credentials(self):
...
-            eda_cred = models.EdaCredential.objects.create(
+            eda_cred, created = models.EdaCredential.objects.get_or_create(
                 name=cred.name,
-                description=cred.description,
-                managed=False,
-                credential_type=cred_type,
-                inputs=inputs_to_store(inputs),
+                defaults={
+                    "description": cred.description,
+                    "managed": True,
```
Comment (hsong-rh) [inline src/aap_eda/core/management/commands/create_initial_data.py:457]: All these converted credentials are user-created, `managed` should be `False`, right?

File: src/aap_eda/core/management/commands/create_initial_data.py [hunk 4]
```diff
@@ -479,17 +492,26 @@ def _copy_scm_credentials(self):
...
```
Comment (hsong-rh) [inline src/aap_eda/core/management/commands/create_initial_data.py:499]: Same question here

File: src/aap_eda/core/management/commands/create_initial_data.py [hunk 5]
```diff
@@ -479,17 +505,26 @@ def _copy_scm_credentials(self):
...
+                models.Project.objects.filter(credential=cred).update(
+                    eda_credential=eda_cred, credential=None
+                )
+                cred.delete()
+            else:
+                warn_msg = (
+                    f"Git Credential {cred.name} already converted to "
+                    "EdaCredential. Skip the duplicated one."
+                )
+                LOGGER.warning(warn_msg)
```
Comment (hsong-rh) [inline src/aap_eda/core/management/commands/create_initial_data.py:527]: Keep them in sync, should we change this into the `info` level too?

Comment (bzwei) [reply src/aap_eda/core/management/commands/create_initial_data.py:527]: sorry, I missed this.

**Review by hsong-rh — APPROVED** [review]

**Review by mkanoor — APPROVED** [review]

**Pull request merged**